src/core/lib/transport/transport.h

```diff
@@ -198,6 +198,8 @@ struct grpc_transport_stream_op_batch_payload {
         grpc_chttp2_grpc_status_to_http2_error. Send a RST_STREAM with this
         error. */
   struct {
+    // Error contract: the recipient of cancel_error must cause cancel_error
```

Comment:
Sounds good.  You could also add a comment on `grpc_transport_stream_op_finish_batch_with_failure()` documenting the fact that it unrefs `cancel_error`.